Refactor code

YoshikiOhtani opened the pull request:
Refactor the code based on the `add-badpixels-func` branch.

YoshikiOhtani commented:
OK so I mainly modified the `MarsCalibratedRun`, the `event_generator` and `parse_simulation_header` functions of the `MAGICEventSource`. I have still some ideas for the refactoring of the other functions, but I should give up here because it takes time and so it's time for reviews. I would appreciate any comments.

YoshikiOhtani commented:
I found some bugs of the bad pixels searches and pushed a new commit

aleberti commented:
Hi, I think that you should rebase your branch now that #31 was merged into master (note that I also added a commit yesterday to fix some MC stuff). In that way I should be able to see the correct diff (now I see both my changes and yours).

YoshikiOhtani commented:
Hi, this pull request is still based on the `add-badpixels-func` branch that is now under reviews, and so once the branch is merged to the master one, I will rebase again this branch.

YoshikiOhtani commented:
Sorry for the long commits but finally I'm satisfied with the refactoring of the code. So eventually I mainly refactored the `MarsCalibratedRun` and `event_generator` of the `MAGICEventSource`. I actually modified the `parse_simulation_config` function but since I didn't understand that Alessio is now implementing the solution to make the module possible to read the other level of MAGIC data, I recovered the original code now.

Regarding the `MarsCalibratedRun`, I simplified the code based on the fact that it reads one calibrated sub-run file. Since one sub-run file is either M1 or M2, I simplified the property function to extract the number of events. Also, I removed the np.concatenate functions.

The tests are passed in local machine, and I will not push any other commits. It would be great if someone could review and merge this if there is no objections. Or please let me know if you have any comments.

aleberti commented on ctapipe_io_magic/__init__.py:
```diff
@@ -961,364 +950,297 @@ def _get_badrmspixel_mask(self, data):
-                pointing.array_ra = np.deg2rad(event_data['pointing_ra'][event_i])*u.rad
-                pointing.array_dec = np.deg2rad(event_data['pointing_dec'][event_i])*u.rad
+                    event.simulation.shower.alt = u.Quantity(90, u.deg) - event_data['mc_theta'][i_event].to(u.deg)
+                    event.simulation.shower.az = u.Quantity(180, u.deg) - event_data['mc_phi'][i_event].to(u.deg) + mfield_dec
```
While the transformation from Corsika to drive frame is fine, can you please re-add the wrapping up to the [-180,180] interval:

```
if true_az > 180:
    true_az -= 360
```

YoshikiOhtani replied:
OK sorry for forgetting this, done.

aleberti commented on ctapipe_io_magic/__init__.py:
```diff
@@ -1345,172 +1267,124 @@ def load_events(uproot_file, is_mc, n_camera_pixels):
-                cut=events_cut[event_type],
-                library="np"
-            )
+                # ToBeChecked: it seems the *.fDev* and *.fRa/fDec branches of MC calibrated
+                # file contain only zeros. If it is the case for all MC files, the following
+                # calculations do not needed and can be simplified:
```
I would leave it, since when real star data are read, this part of code will be used

YoshikiOhtani replied:
OK thanks for your information, done.

YoshikiOhtani commented:
OK so I have modified the part that @aleberti mentioned.

aleberti commented:
Ok, we can merge! Also, I will check the part about bad pixels to cross-check that the resulting images are the same (as you did).

YoshikiOhtani commented:
Thanks a lot!
